activity/coordinated_worker.go

```diff
@@ -17,101 +17,105 @@ const (
 
 func (w *ActivityWorker) AddCoordinatedHandler(heartbeatInterval time.Duration, handler *CoordinatedActivityHandler) {
 
-	ticks := make(chan CoordinatedActivityHandlerTick)
-	startErr := make(chan error)
+	handlerFunc := func(ctx *ActivityContext, input interface{}) {
+		ticks := make(chan CoordinatedActivityHandlerTick)
+		startErr := make(chan error)
 
-	handlerTickFunc := func(activityTask *swf.ActivityTask, input interface{}) {
-		go func() {
-			cont, res, err := handler.Tick(activityTask, input)
-			ticks <- CoordinatedActivityHandlerTick{
-				Continue: cont,
-				Result:   res,
-				Error:    err,
-			}
-		}()
-	}
+		handlerTickFunc := func(cxt *ActivityContext, input interface{}) {
+			go func() {
+				cont, res, err := handler.Tick(cxt, input)
+				ticks <- CoordinatedActivityHandlerTick{
+					Continue: cont,
+					Result:   res,
+					Error:    err,
+				}
+			}()
+		}
 
-	handlerStartFunc := func(activityTask *swf.ActivityTask, input interface{}) {
-		go func() {
-			update, err := handler.Start(activityTask, input)
-			if err != nil {
-				startErr <- err
-			} else {
-				err = w.signalStart(activityTask, update)
+		handlerStartFunc := func(ctx *ActivityContext, input interface{}) {
+			go func() {
+				update, err := handler.Start(ctx, input)
 				if err != nil {
 					startErr <- err
 				} else {
-					handlerTickFunc(activityTask, input)
+					err = w.signalStart(ctx, update)
+					if err != nil {
+						startErr <- err
+					} else {
+						handlerTickFunc(ctx, input)
+					}
 				}
-			}
-		}()
-	}
+			}()
+		}
 
-	heartbeats := time.Tick(heartbeatInterval)
+		heartbeats := time.Tick(heartbeatInterval)
```

Comment:
I believe you probably should use `heartbeats := time.NewTicker(heartbeatInterval)` and `defer heartbeats.Stop()` to avoid leaking ticker goroutines in here.